pkg/module/remote.go

```diff
@@ -38,7 +39,18 @@ func Push(archive *ctf.ComponentArchive, r *Remote, log *zap.SugaredLogger) erro
 
 	// update repository context
 	if len(r.Registry) != 0 {
-		if err := cdv2.InjectRepositoryContext(archive.ComponentDescriptor, cdv2.NewOCIRegistryRepository(r.Registry, cdv2.OCIRegistryURLPathMapping)); err != nil {
+		nameMapping := cdv2.OCIRegistryURLPathMapping
+		if rc := archive.ComponentDescriptor.GetEffectiveRepositoryContext(); rc != nil {
```

Comment:
Thanks for catching that. Mechanical refactoring resulted in the code, that now looks a bit strange :)
I am getting the a value from the repositoryContext just to build a new object using this value and inject it back into the same repositoryContext... It may be redundant now, I will double-check.

Reply:
Fixed